### PR TITLE
feat: add usePreviousDistinct

### LINKED ISSUE #476: usePrevious only tracks the previous of the last render and is affected by other hooks
If i have other hooks in a component that are unrelated to what the `usePrevious` is tracking, some update that causes a rerender from that hook will sync the previous value to the current value. To see an example of this, here's a [sandbox](https://codesandbox.io/s/react-use-useprevious-0s9po). To see the behavior, click `+` a few times and then click `toggle on` to toggle unrelated state. Not that the value for `prev` will update.

In this sandbox I've implemented an alternative usePrevious that tracks a more accurate version of previous.

I can make a PR but wanted to know if maybe we call this new version something else to support the previous version of `usePrevious`.

I understand that `usePrevious` tracks the previous render value, but this version may be more strictly useful in an application. Maybe `usePreviousDistinct`?

### PR DESCRIPTION
**Pull request by psachs21**
Fixes #476

### CODE REVIEW

File: src/__tests__/usePreviousDistinct.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { cleanup, renderHook } from 'react-hooks-testing-library';
+import usePrevious from '../usePreviousDistinct';
+
+afterEach(cleanup);
```
Comment (Belco90) [inline src/__tests__/usePreviousDistinct.test.tsx]: Cleanup not necessary anymore, so please remove this line

File: src/__tests__/usePreviousDistinct.test.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { cleanup, renderHook } from 'react-hooks-testing-library';
```
Comment (Belco90) [inline src/__tests__/usePreviousDistinct.test.tsx]: react-hooks-testing-library was updated to scoped version, so you need to import it from @testing-library/react-hooks

Comment (psachs21) [reply src/__tests__/usePreviousDistinct.test.tsx]: Ah, haven't updated to latest in a bit. Will do so.

**Comment by Belco90** [comment]
Looks good to me! Let's see what @xobotyi @streamich think.

**Comment by xobotyi** [comment]
LGTM too

**Comment by psachs21** [comment]
> Nice hook! It looks good to me, but it would need an example file for storybook and some changes on test file.
> 
> EDIT: sorry, just realize the example is already there 💯

No, this was a good catch. I realize i hadn't even run storybook to check if the docs were picked up. I've push an update with the docs.

**Pull request merged**